src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -93,7 +93,7 @@ public function onKernelRequest(RequestEvent $event): void
      */
     public function onKernelResponse(ResponseEvent $event): void
     {
-        if (!$event->isMainRequest() || (!$this->container->has('initialized_session') && !$event->getRequest()->hasSession())) {
```

Comment:
initialized_session is gone
hasSession is checked below

Reply:
`hasSession()` is called with `true` below. Does that make a difference?

Reply:
It doesn't to me. Both calls are side-effect free and the second one is stricter.